source/s3_auto_ranged_put.c

```diff
@@ -988,6 +999,17 @@ static int s_s3_auto_ranged_put_prepare_request(
         }
         case AWS_S3_AUTO_RANGED_PUT_REQUEST_TAG_COMPLETE_MULTIPART_UPLOAD: {
 
+            if (!auto_ranged_put->has_content_length &&
+                auto_ranged_put->prepare_data.num_parts_read_from_stream == 0) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_S3_META_REQUEST,
+                    "id=%p Could not create complete multipart upload for meta request with zero content length.",
```

Comment:
Error scolds the user for choosing to use Multipart Upload, but that's an implementation detail.

The user chose not to specify Content-Length, so let's phrase the error around that

Maybe something like:
> Set "Content-Length: 0" to upload an empty object. 0 byte meta requests lacking a Content-Length header are not currently supported.

Reply:
Add add code comment about how CompleteMultipartUpload will error if there are no parts

We can ultimately support this when we shift to buffering the first part to determine whether MPU is necessary